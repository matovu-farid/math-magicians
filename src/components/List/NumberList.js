import propTypes from 'prop-types';
import { Component } from 'react';
import { NumberTile } from '../Tiles/Tile';

class NumberList extends Component {
  constructor(props) {
    super(props);
    this.state = {};
  }

  render() {
    const { handleButtonClick } = this.props;
    return (
      <ul className="num-list">

        {
      Array.from(Array(9).keys()).map((i) => <NumberTile key={i} handleButtonClick={handleButtonClick} value={`${i + 1}`} />)
    }
      </ul>
    );
  }
}

NumberList.propTypes = {
  handleButtonClick: propTypes.func.isRequired,
};

export default NumberList;
