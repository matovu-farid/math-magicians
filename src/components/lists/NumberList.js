import { Component } from 'react';
import { NumberTile } from '../Tiles/Tile';

class NumberList extends Component {
  constructor(props) {
    super(props);
    this.state = {};
  }

  render() {
    return (
      <ul className="num-list">

        {
      Array.from(Array(9).keys()).map((i) => <NumberTile key={i} value={`${i + 1}`} />)
    }
      </ul>
    );
  }
}

export default NumberList;
