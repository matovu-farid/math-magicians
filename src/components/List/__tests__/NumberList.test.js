import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';
import NumberList from '../NumberList';

it('should have 9 list items', () => {
  render(<NumberList />);
  const listItems = screen.queryAllByRole('listitem');

  expect(listItems.length).toBe(9);
});
