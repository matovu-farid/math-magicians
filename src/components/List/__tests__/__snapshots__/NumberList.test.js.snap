// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<ul
  className="num-list"
>
  <li>
    <button
      className="tile"
      onClick={[Function]}
      type="button"
    >
      1
    </button>
  </li>
  <li>
    <button
      className="tile"
      onClick={[Function]}
      type="button"
    >
      2
    </button>
  </li>
  <li>
    <button
      className="tile"
      onClick={[Function]}
      type="button"
    >
      3
    </button>
  </li>
  <li>
    <button
      className="tile"
      onClick={[Function]}
      type="button"
    >
      4
    </button>
  </li>
  <li>
    <button
      className="tile"
      onClick={[Function]}
      type="button"
    >
      5
    </button>
  </li>
  <li>
    <button
      className="tile"
      onClick={[Function]}
      type="button"
    >
      6
    </button>
  </li>
  <li>
    <button
      className="tile"
      onClick={[Function]}
      type="button"
    >
      7
    </button>
  </li>
  <li>
    <button
      className="tile"
      onClick={[Function]}
      type="button"
    >
      8
    </button>
  </li>
  <li>
    <button
      className="tile"
      onClick={[Function]}
      type="button"
    >
      9
    </button>
  </li>
</ul>
`;
